File: Client/Aktivnost.js
import { Clan } from "./Clan.js";
import { Padobran } from "./Padobran.js";
import { Avion } from "./Avion.js";

export class Aktivnost{

    constructor(id, vreme, datum, visina, clanid, avionid, padobranid){
        this.id=id;
        this.vreme=vreme;
        this.datum=datum;
        this.visina=visina;
        this.clan=clanid;
        this.padobran=padobranid;
        this.avion=avionid;
    }

    crtaj(host){

        var tr=document.createElement("tr");
        host.appendChild(tr);

        var el=document.createElement("td");
        el.innerHTML=this.vreme;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.datum;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.visina;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.clan;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.avion;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.padobran;
        tr.appendChild(el);
    }
}

File: Models/Aktivnost.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Models
{
    [Table("Aktivnost")]
    public class Aktivnost
    {
        [Key]
        public int ID { get; set; }

        //[RegularExpression("(1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm])")]
        [Required]
        public string Vreme { get; set; }

        //[RegularExpression("((0[1-9]|[12][0-9]|30)[-/]?(0[13-9]|1[012])|31[-/]?(0[13578]|1[02])|(0[1-9]|1[0-9]|2[0-8])[-/]?02)[-/]?[0-9]{4}|29[-/]?02[-/]?([0-9]{2}(([2468][048]|[02468][48])|[13579][26])|([13579][26]|[02468][048]|0[0-9]|1[0-6])00)")]
        [Required]
        public string Datum { get; set; }

        [Required]
        [Range(1000,4000)]
        public int Visina { get; set; }

        public Clan Clan { get; set; }
        
        public Avion Avion { get; set; }

        public Padobran Padobran { get; set; }
    }
}